src/texas_host/core.clj

```diff
@@ -1,27 +1,27 @@
 (ns texas-host.core
   (:gen-class))
 
-(defn remove-card [card pack]
+(defn remove-card [card deck]
   (remove
     (fn [x] (= x card))
-    pack))
+    deck))
 
-(defn get-card [pack]
-  ( let [card (rand-nth pack)]
-  [card (remove-card card pack)]))
+(defn choose-card [deck]
+  [(rand-nth deck)])
 
-(defn get-cards [amount cards pack]
+(defn deal-cards [amount deck]
   (cond
-    (= amount 0) [cards]
-    :else (let [[new-card new-pack] (get-card pack)]
-                   (get-cards (- amount 1) (conj cards new-card) new-pack))))
-
-(defn deal-cards [amount pack]
-  (get-cards amount [] pack))
+    (= amount 0) [[] deck]
+    :else
+    (let [[cards-before deck-before] (deal-cards (- amount 1) deck)]
```

Comment:
While amount is greater than zero, deal-cards function is called recursively. When amount is zero, an empty vector is returned. In the previous function call a new card is chosen and added to the cards dealt. And so on and so on until the cards dealt are returned from the first deal-cards function call.